workspaces/ui-v2/src/lib/Interfaces.ts

```diff
@@ -198,45 +198,25 @@ export const DiffInResponse = (key: string): boolean =>
 
 // The ones we like to work with in the UI
 
-// TODO QPB remove these external exports
-export interface IRequestBodyLocation {
-  contentType?: string;
-  requestId?: string;
-}
-
-export interface IResponseBodyLocation {
-  statusCode: number;
-  contentType?: string;
-  responseId?: string;
-}
-
-// TODO QPB - refactor this type such that
-/**
- * 
 export interface IParsedLocation {
   pathId: string;
   method: string;
-  location: {
-    type: 'query'
-  } | {
-    type: 'request'
-    requestId: string
-    contentType: string
-  } | {
-    type: 'response'
-    statusCode: number
-    contentType: string
-    responseId: string
-  }}
- */
-export interface IParsedLocation {
-  pathId: string;
-  method: string;
-  inQuery?: {
-    queryParametersId: string;
-  };
-  inRequest?: IRequestBodyLocation;
-  inResponse?: IResponseBodyLocation;
+  data:
+    | {
+        type: 'query';
+        queryParametersId: string;
+      }
+    | {
+        type: 'request';
+        requestId: string;
+        contentType: string;
+      }
+    | {
+        type: 'response';
+        statusCode: number;
+        contentType: string;
+        responseId: string;
+      };
```

Comment:
👍 we should definitely not be too afraid to use anonymous types where appropriate!